pkg/controller/machine-set-boot-image/machine_set_boot_image_controller.go

```diff
@@ -518,6 +362,50 @@ func (ctrl *Controller) syncMachineResource(key string) error {
 		return fmt.Errorf("failed to fetch infra object during machineset sync: %w", err)
 	}
 
+	// Wait until the MCO hash version stored in the configmap matches the current MCO
+	// version. This is done by the operator when a master node successfully updates to a new image. This is
+	// to prevent machinesets from being updated before the operator itself has updated.
+	// Could return an error(and cause degrade) immediately here, but seems excessive. Waiting with a timeout
+	// is a bit more graceful.
+	var configMap *corev1.ConfigMap
+	var pollError error
+	if err = wait.PollUntilContextTimeout(context.TODO(), 1*time.Minute, 15*time.Minute, true, func(ctx context.Context) (bool, error) {
+		klog.Infof("Waiting until coreos-bootimages config map has been stamped by the current version of the operator")
```

Comment:
nit: maybe log before we start the poll instead of every loop. (Alternatively I think the pollErrors probably already cover what we want to convey)